mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/massindexing/spi/PojoMassIndexerAgent.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.pojo.massindexing.spi;
+
+import java.util.concurrent.CompletableFuture;
+
+public interface PojoMassIndexerAgent {
+
+	static PojoMassIndexerAgent noOp() {
+		return NoOpPojoMassIndexerAgent.INSTANCE;
+	}
+
+	/**
+	 * Starts requesting other agents from other agents that could possibly perform indexing (e.g. automatic indexing)
+	 * that they suspend themselves.
+	 * <p>
+	 * Other agents can be considered suspended when the returned future completes successfully;
+	 * they will remain suspended until this agent is {@link #preStop() pre-stoppped} or {@link #stop() stopped}.
+	 *
+	 * @return A future that completes successfully when other agents have been successfully suspended.
+	 * If no agents can be suspended (e.g. no coordination), returns a successfully completed future immediately.
```

Comment:
When there is no coordination, there are still "agents" of sorts, but those are each thread that performs database changes; the thread will update the index itself. In that case, we simply *cannot* suspend those agents, because they are controlled by the user.

So, no, here I really meant "can".